src/App.tsx

```diff
@@ -1,12 +1,79 @@
 import React from 'react';
+import { GoodsList } from './components/GoodsList';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[],
+  showGoods: Good[],
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+    showGoods: [],
+  };
+
+  async componentDidMount() {
+    const goods = await getAll();
+
+    this.setState({ goods });
+  }
+
+  allGoods = () => {
+    this.setState(state => ({
+      showGoods: state.goods,
+    }));
+  };
+
+  firstFive = () => {
+    this.setState(state => ({
+      showGoods: get5First(state.goods),
+    }));
+  };
+
+  redGoods = () => {
+    this.setState(state => ({
+      showGoods: getRedGoods(state.goods),
+    }));
+  };
```

Comment:
create one function for this which will accept the callback (get5First, getAll, getRedGoods) and write the result of it to state